examples/performance-debugging/performance-debugging.cpp

```diff
@@ -0,0 +1,500 @@
+        // Generate and call apply on a solver
+        solver_factory->generate(A)->apply(lend(b), lend(x_clone));
+        exec->synchronize();
+    }
+
+    // Do a timed run
+    {
+        // Clone x to not overwrite the original one
+        auto x_clone = clone(x);
+
+        // Synchronize ensures no operation are ongoing
+        exec->synchronize();
+        // Time before generate
+        auto g_tic = std::chrono::steady_clock::now();
+        // Generate a solver
+        auto generated_solver = solver_factory->generate(A);
+        exec->synchronize();
+        // Time after generate
+        auto g_tac = std::chrono::steady_clock::now();
+        // Compute the generation time
+        auto generate_time =
+            std::chrono::duration_cast<std::chrono::nanoseconds>(g_tac - g_tic);
+        // Write the generate time to the output file
+        output_file << "Generate time (ns): " << generate_time.count()
+                    << std::endl;
+
+        // Similarly time the apply
+        exec->synchronize();
+        auto a_tic = std::chrono::steady_clock::now();
+        generated_solver->apply(lend(b), lend(x_clone));
+        exec->synchronize();
+        auto a_tac = std::chrono::steady_clock::now();
+        auto apply_time =
+            std::chrono::duration_cast<std::chrono::nanoseconds>(a_tac - a_tic);
+        output_file << "Apply time (ns): " << apply_time.count() << std::endl;
+
+        // Compute the residual norm
+        auto residual =
+            utils::compute_residual_norm(lend(A), lend(b), lend(x_clone));
+        output_file << "Residual_norm: " << residual << std::endl;
+    }
+
+    // Log the internal operations using the OperationLogger without timing
+    {
+        // Clone x to not overwrite the original one
+        auto x_clone = gko::clone(x);
+
+        // Create an OperationLogger to analyze the generate step
+        auto gen_logger = std::make_shared<loggers::OperationLogger>(exec);
+        // Add the generate logger to the executor
+        exec->add_logger(gen_logger);
+        // Generate a solver
+        auto generated_solver = solver_factory->generate(A);
+        // Remove the generate logger from the executor
+        exec->remove_logger(gko::lend(gen_logger));
+        // Write the data to the output file
+        output_file << "Generate operations times (ns):" << std::endl;
+        gen_logger->write_data(output_file);
+
+        // Create an OperationLogger to analyze the apply step
+        auto apply_logger = std::make_shared<loggers::OperationLogger>(exec);
+        exec->add_logger(apply_logger);
+        // Create a ResidualLogger to log the recurent residual
+        auto res_logger = std::make_shared<loggers::ResidualLogger<vtype>>(
+            exec, gko::lend(A), gko::lend(b));
+        generated_solver->add_logger(res_logger);
+        // Solve the system
+        generated_solver->apply(lend(b), lend(x_clone));
+        exec->remove_logger(gko::lend(apply_logger));
```

Comment:
In these two lines, you mention `gko::` in the second line, but not in the first.

Reply:
Thanks, I actually found from the other examples that none of these had the `gko::` prefix, (see preconditioned-solver example for example). I tried to add `gko::` at least in this one to the copy/pasted lines, but looks like I missed a lot of them.